vip-helpers/vip-caching.php

```diff
@@ -710,15 +710,17 @@ function wpcom_vip_flush_wp_old_slug_redirect_cache( $post_id, $post, $post_befo
  * We're seeing an increase of urls that match this pattern: http://example.com/http://othersite.com/random_text
  * These then cause really slow lookups inside of wp_old_slug_redirect, since wp_old_slug redirect does not match on full urls but rather former slugs it's safe to skip the lookup for these. (Most of the calls are from bad ad providers that generate random urls)
  */
-function wpcom_vip_maybe_skip_old_slug_redirect(){
+function wpcom_vip_maybe_skip_old_slug_redirect() {
+	$uri = $_SERVER['DOCUMENT_URI'] ?? $_SERVER['REQUEST_URI'] ?? '';
 
 	//We look to see if a malformed url (represented by 'http:' ) is right after the starting / in DOCUMENT_URI hence position 1
-	if ( is_404() && ( 1 === strpos( $_SERVER['DOCUMENT_URI'], 'http:' ) || 1 === strpos( $_SERVER['DOCUMENT_URI'], 'https:' ) ) ) {
+	if ( is_404() && ( 1 === strpos( $uri, 'http:' ) || 1 === strpos( $uri, 'https:' ) ) ) {
```

Comment:
Since you're comparing to e.g.: http://othersite.com/random_text and looking to match the start of the string, this should be comparisons to 0 since string indexes start at 0.
```suggestion
	if ( is_404() && ( 0 === strpos( $uri, 'http:' ) || 0 === strpos( $uri, 'https:' ) ) ) {
```

Reply:
@brandon-m-skinner How does that match up with the comment then?

```
//We look to see if a malformed url (represented by 'http:' ) is right after the starting / in DOCUMENT_URI hence position 1
```

As per the function DocBlock, what we're looking for is an invalid URL e.g. `http://example.com/http://othersite.com/random_text`, which means the DOCUMENT_URI and REQUEST_URI would be `/http:...`, so looking for `http:` or `https:` starting at index 1.

We could just start at index 0 and look for `/http:` or `/https:` if that makes it easier to understand? (I think it would be)

Also, do we really need to check for DOCUMENT_URI if REQUEST_URI seems to be the most popular key to check? Compare [instances of DOCUMENT_URI](https://github.com/Automattic/vip-go-mu-plugins/search?q=DOCUMENT_URI) to [instances of REQUEST_URI](https://github.com/Automattic/vip-go-mu-plugins/search?q=REQUEST_URI) just in this repo.